src/main/java/io/nats/client/api/ConsumerConfiguration.java

```diff
@@ -301,7 +319,7 @@ public static Builder builder(ConsumerConfiguration cc) {
         private long maxAckPending = 0;
         private Duration idleHeartbeat = Duration.ZERO;
         private boolean flowControl;
-        private long maxPullWaiting = DEFAULT_MAX_PULL_WAITING;
+        private long maxPullWaiting = 0;
```

Comment:
0 is a better default than the constant which is 512, mostly since we don't send this to the server if it's 0 or the default. Plus I suppose the default could change on the server at some point